contrib/database/sql/conn.go

```diff
@@ -167,8 +179,28 @@ func WithSpanTags(ctx context.Context, tags map[string]string) context.Context {
 	return context.WithValue(ctx, spanTagsKey, tags)
 }
 
+// injectComments returns the query with sql comments injected according to the comment injection mode along
+// with a span id injected into sql comments. The returned span id should be used when the sql span is created
+// following the traced database call.
```

Comment:
```suggestion
// injectComments returns the query with SQL comments injected according to the comment injection mode along
// with a span ID injected into SQL comments. The returned span ID should be used when the SQL span is created
// following the traced database call.
```
I would phrase this a bit differently, perhaps making it more clear (IMHO! I don't feel strongly on it):
```go
// injectComments injects any tracing context found in ctx into query based on the set mode. It returns the 
// query with the comments injected. If there is no span in ctx, it will generate a spanID and return it so that
// it can be used to create the span that represents the DB call using this query.
```
I think that the returned `spanID` should likely be `0` if there is a span in the context, or that there should be a third `ok` argument which indicates that. The next span should be a child of whatever was in `ctx` and not use the same `SpanID`. A duplicate `SpanID` in a trace will determine the agent to drop the entire trace.

We should test for this case...

Reply:
This ties in with this [other thread](https://github.com/DataDog/dd-trace-go/pull/1226/files#r894179529). 

It wouldn't be accurate to say we only return a span id when there's no span context because we always need to create one. In the case where there's an existing `span`/`trace`, that `spanID` is the parent's `spanID` and not the `spanID` for the SQL span. That's because the SQL span gets created _after_ the database call. 

That's the main reason we have to generate a `spanID` during injection and I don't really see a way around it until we start following the start/finish span which would only be possible if we were okay creating spans even in cases of `driver.ErrSkip`. Or if we had a `span.Cancel()` to allow a span to be cancelled after it's created when we find calls that returned `driver.ErrSkip`.